be/src/resolvers/dbpedia-resolve.service.ts

```diff
@@ -1,22 +1,129 @@
     }
 
+    const url: URL = new URL(uri.toString().replace('http:', 'https:').replace('resource', 'data') + '.ttl')
+    
+    const store: Store = $rdf.graph();
+    const fetcher = new $rdf.Fetcher(store, {timeout: 5000});
+
+    const response = {}
+
+    const subject = $rdf.sym(uri.toString())
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    // @ts-ignore
+    return fetcher.load(url.toString(), {}).then(() => {
+
+      const keys = Object.keys(this.props)
+      // console.log(keys)
+
+      /* eslint-disable */
+      keys.forEach((key) => {
+        // @ts-ignore
+        for (let i = 0; i < this.props[key].length; i++){
+          // @ts-ignore
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+          const curr = store.match(subject, $rdf.sym(this.props[key][i]))
+              .map(q => q.object)
+              .filter(o => o instanceof Literal)
+              // @ts-ignore
+              .filter((l: Literal) => l.language == this.language)
+          if (!!curr){
+            // @ts-ignore
+            response[`${key}`] = curr.toString()
+            break
+          }
+        }
+      });
+
+      /* eslint-enable */
+
+
+
+      /** Represents all the possible predicates on the given subject */
+
+      // store.match(subject)
+      //   .map(q => q.predicate)
+      //   .map(p => p.value)
+      //   .sort()
+      //   .reduce(((previousValue, currentValue) => {
+      //     if (previousValue !== currentValue) {
+      //       console.log(currentValue)
+      //     }
+      //     return currentValue
+      //   })
+      //   )
+
+      /** Shows all the possible predicates with all languages */
+
+      // for (let i = 0; i < this.props.title.length; i++) {
+      //   console.log('\n')
+      //   console.log(this.props.title[i])
+      //   store.match(subject, $rdf.sym(this.props.title[i]))
+      //     .map(q => q.object)
+      //     .filter(o => o instanceof Literal)
+      //   // .filter((l: Literal) => l.language == language)
+      //   // .map(p => p.value)
+      //   // .sort()
+      //   //   .reduce(((previousValue: Literal, currentValue: Literal) => {
+      //   //     // console.log('previous: ', previousValue)
+      //   //     // console.log('current: ', currentValue)
+      //   //     if (previousValue !== currentValue) {
+      //   //       console.log(currentValue.language + ': ' + currentValue.value)
+      //   //     }
+      //   //     return currentValue
+      //   //   })
+      //   //   )
+      //     .forEach((res ) => {
+      //       // console.log('size: ', array.length)
+      //       console.log(res.value)
+      //     })
+      // }
+
+      // running = false
+      return Promise.resolve({data: response})
+    }
+    ).catch(e => 
+      // console.error(e);
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+      ServiceErrorFactory.requestTimeout(e.toString())
```

Comment:
```suggestion
return ServiceErrorFactory.requestTimeout(e.toString());
```
to reject the Promise and communicate the error, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch

Reply:
It doesn't let write return before the Error and insists on keeping like that.

Reply:
There are two ways to write this:
```
.catch(e => ServiceErrorFactory.requestTimeout(e.toString()))
```
or
```
.catch(e => { return ServiceErrorFactory.requestTimeout(e.toString()); })
```